pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,787 @@
+
+    @property
+    def http_proxy_password_set(self):
+        """Report if the HTTP proxy password has been set.
+
+        We need this separate property, because the
+        Subscription DBUS module API on purpose does
+        not provide the plaintext of the HTTP proxy password
+        once it has been set. Therefore we only know
+        a password has been set via kickstart or
+        the DBUS API, but can't get its plaintext.
+
+        Thus we have this property to check if a HTTP proxy
+        password has been set.
+        """
+        return self.http_proxy_password or self._subscription_module.proxy.ServerProxyPasswordSet
+
+    @property
+    def show_http_proxy_password_placeholder(self):
+        """Show a placeholder on the HTTP proxy password field.
+
+        The placeholder notifies the user about HTTP proxy password
+        being set in the DBus module.
+
+        The placeholder will be only shown if there is no
+        actual text in the entry field.
+        """
+        return bool(self._http_proxy_password_entry.get_placeholder_text())
+
+    @show_http_proxy_password_placeholder.setter
+    def show_http_proxy_password_placeholder(self, show_placeholder):
+        if show_placeholder:
+            self._http_proxy_password_entry.set_placeholder_text(_("Password set."))
+        else:
+            self._http_proxy_password_entry.set_placeholder_text("")
+
+    @property
+    def http_proxy_set(self):
+        # we consider the HTTP proxy data as set only if the corresponding checkbox is checked
+        return self._http_proxy_checkbox.get_active()
+
+    @property
+    def custom_rhsm_baseurl(self):
+        return self._custom_rhsm_baseurl_entry.get_text()
+
+    @custom_rhsm_baseurl.setter
+    def custom_rhsm_baseurl(self, baseurl):
+        self._custom_rhsm_baseurl_entry.set_text(baseurl)
+
+    @property
+    def custom_rhsm_baseurl_set(self):
+        # we consider the custom RHSM baseurl as set only if the corresponding checkbox is checked
+        return self._custom_rhsm_baseurl_checkbox.get_active()
+
+    @property
+    def username(self):
+        """Red Hat account login."""
+        return self._custom_server_hostname_entry.get_text()
+
+    @username.setter
+    def username(self, username):
+        self._username_entry.set_text(username)
+        self.authentication_data_changed.emit()
+
+    @property
+    def password(self):
+        """Red Hat account password."""
+        return self._password_entry.get_text()
+
+    @password.setter
+    def password(self, password):
+        self._password_entry.set_text(password)
+        self.authentication_data_changed.emit()
+
+    @property
+    def organization(self):
+        """Organization name."""
+        return self._organization_entry.get_text()
+
+    @organization.setter
+    def organization(self, organization):
+        self._organization_entry.set_text(organization)
+        self.authentication_data_changed.emit()
```

Comment:
I think that you could have a callback that will emit the `authentication_data_changed` signal when a widget is changed and register the callback in the `.glade` file in all related widgets. This way we have about 200-300 lines of boilerplate.

Reply:
I don't think we actually need this signal as long as the relevant data in the Subscription DBus module is not changed after kickstart is parsed by other actors than the Subscription spoke itself.

So I'll just drop that signal & make sure registration update is run at the end of `initialize`.